splash/network_manager.py

```diff
@@ -195,11 +195,12 @@ def _handle_custom_headers(self, request):
             request.setRawHeader(name, value)
 
     def _handle_request_cookies(self, request):
-        jar = QNetworkCookieJar()
-        self.setCookieJar(jar)
         cookiejar = self._get_webpage_attribute(request, "cookiejar")
         if cookiejar is not None:
             cookiejar.update_cookie_header(request)
+            self.setCookieJar(cookiejar)
```

Comment:
Hmm, who owns/deletes cookiejars? 

A cookiejar is created in SplashQWebPage. Then it get re-parented to our network manager (according to setCookieJar docs). But we may have several web pages alive (one per slot), but only a single network access manager. When you set a new cookiejar using setCookieJar old cookiejar is deleted (see https://github.com/qtproject/qtbase/blob/bafbf3b872103409370fbda43a2ce3ad153a33f4/src/network/access/qnetworkaccessmanager.cpp#L677). I wonder if it may cause https://github.com/scrapinghub/splash/issues/321.